ddp-workspace/projects/ddp-sdk/src/lib/models/activity/activityBlock.ts

```diff
@@ -8,6 +8,11 @@ export abstract class ActivityBlock {
 
     abstract get blockType(): BlockType;
 
+    /**
+     * all block contained by this block, including self
+     */
+    abstract get blocks(): Array<ActivityBlock>;
```

Comment:
Force every implementation to return blocks under it's control, including itself. Sets up a way to recurse through all the blocks.